src/lib/message-router.js

```diff
@@ -105,7 +106,25 @@ MessageRouter.prototype._handleRequest = function * (request, sender) {
  * @param {Route[]} routes
  * @param {IlpAddress} sender
  */
-MessageRouter.prototype.receiveRoutes = function * (routes, sender) {
+MessageRouter.prototype.receiveRoutes = function * (payload, sender) {
+  debug('receiveRoutes sender:', sender)
+  let routes = payload.new_routes
+
+  let holdDownTime = payload.hold_down_time
+  this.routingTables.bumpConnector(sender, holdDownTime)
+  let potentiallyUnreachableLedgers = payload.unreachable_through_me
+  let lostLedgerLinks = []
+  if (potentiallyUnreachableLedgers.length > 0) {
+    log.info('informed of broken routes to:', potentiallyUnreachableLedgers, ' through:', sender)
+    for (const ledger of potentiallyUnreachableLedgers) {
+      lostLedgerLinks.concat(this.routingTables.invalidateConnectorsRoutesTo(sender, ledger))
```

Comment:
This is a noop; concat doesn't modify either array (and there should be tests to catch that as well).